homeassistant/components/omnilogic/light.py

```diff
@@ -0,0 +1,225 @@
+
+            self._state = (
+                int(self.coordinator.data[self._item_id][self._state_key]) != 0
+            )
+
+        return self._state
+
+    @property
+    def effect(self):
+        """Return the current light effect."""
+        effect = LightEffect(self.coordinator.data[self._item_id]["currentShow"])
+        return effect.name
+
+    @property
+    def effect_list(self):
+        """Return the supported light effects."""
+        if self._version == 2:
+            return list(LightEffect.__members__)
+
+        return list(LightEffect.__members__)[:17]
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features of the light."""
+        return SUPPORT_EFFECT
+
+    async def async_set_effect(self, effect):
+        """Set the light show effect."""
+        success = await self.coordinator.api.set_lightshow(
+            int(self._item_id[1]),
+            int(self._item_id[3]),
+            int(self._item_id[-1]),
+            int(LightEffect[effect].value),
+        )
+
+        if success:
+            self.async_write_ha_state()
+
+    async def async_turn_on(self, **kwargs):
+        """Turn on the light."""
+        self._state = True
+        self._last_action = time.time()
+        self.async_write_ha_state()
+
+        if kwargs.get(ATTR_EFFECT):
+            await self.async_set_effect(kwargs[ATTR_EFFECT])
+
+        await self.coordinator.api.set_relay_valve(
+            int(self._item_id[1]),
+            int(self._item_id[3]),
+            int(self._item_id[-1]),
+            1,
+        )
+
+    async def async_turn_off(self, **kwargs):
+        """Turn off the light."""
+        self._state = False
+        self._last_action = time.time()
+        self.async_write_ha_state()
+
+        await self.coordinator.api.set_relay_valve(
+            int(self._item_id[1]),
+            int(self._item_id[3]),
+            int(self._item_id[-1]),
+            0,
+        )
+
+    async def async_set_v2effect(self, **kwargs):
+        """Set the light effect speed or brightness for V2 lights."""
+
+        if self._version == 2:
+            speed = kwargs.get("speed", self._speed)
+            brightness = kwargs.get("brightness", self._brightness)
+            if 0 <= speed <= 8 and 0 <= brightness <= 4:
+                await self.coordinator.api.set_lightshowv2(
+                    int(self._item_id[1]),
+                    int(self._item_id[3]),
+                    int(self._item_id[-1]),
+                    int(self.coordinator.data[self._item_id]["currentShow"]),
+                    speed,
+                    brightness,
+                )
+
+            else:
+                raise OmniLogicException("Speed must be 0-8 and brightness 0-4.")
+        else:
+            raise OmniLogicException(
+                "Cannot set effect speed or brightness on version 1 lights."
+            )
```

Comment:
Same as above, if you change it to this, and move it up, you can un-indent this all.

```suggestion
        if self._version != 2:
            raise OmniLogicException(
                "Cannot set effect speed or brightness on version 1 lights."
            )
```

Reply:
Not anywhere near as valueable input as frenck's, but one food for thought.  Depending on how this is refactored, it might make sense to make the message be related version 2.  e.g. to avoid confusion in the event that version 3 lights are one day released (which would be treated as V1 "by default").

One possible way to adjust the message: "Light effect speed and brightness only supported with version 2 lights."

The wording is also a hint that the h/w once it is imagined and produced may support, but the code needs updates.

Reply:
Have changed to a guard @frenck 

@edrikk if they release a lights version 3 we will need to actually do some work in the Python library to extract any of that new functionality so I would address that at that stage. I have found it very hard to "future-proof" anything related to this integration.